lib/Runtime/HostManager/HostManager.cpp

```diff
@@ -269,10 +269,48 @@ llvm::Error HostManager::runNetworkBlocking(llvm::StringRef networkName,
   return std::move(*DCHECK_NOTNULL(runErr.get()));
 }
 
+void HostManager::dispatchRun(InferRequest request) {
+  executor_->run(
+      networks_[request.networkName].dag.root.get(), std::move(request.context),
+      request.requestID,
+      [this, callback = request.callback,
+       name = request.networkName](RunIdentifierTy runID, llvm::Error err,
+                                   std::unique_ptr<ExecutionContext> context) {
+        {
+          std::lock_guard<std::mutex> networkLock(networkLock_);
+          auto it = networks_.find(name);
+          if (it != networks_.end()) {
+            it->second.refcount--;
+          }
+        }
+        TRACE_EVENT_INSTANT(context->getTraceContext(), TraceLevel::RUNTIME,
+                            "finish_" + name);
+        callback(runID, std::move(err), std::move(context));
+
+        // if there is work in the stack pop the next item and kick
+        // it off.
+        {
+          std::lock_guard<std::mutex> networkLock(networkLock_);
+          if (inferQueue_.size()) {
+            // Get the next request, unfortunately priority_queue only provides
+            // a const ref to the top element, since we need to move it we first
+            // cast it to remove the const.
+            auto request =
+                std::move(const_cast<InferRequest &>(inferQueue_.top()));
+            inferQueue_.pop();
+            dispatchRun(std::move(request));
```

Comment:
When a request is pop from the queue, do we need to add 1 to  activeRequestCount_? Or it will be added somewhere?

Reply:
activeRequest gets decremented if there are no more requests in the queue, but queuedRequestCount_ definitely needs to be decremented.